File: client/lib/data/dto/group_setting_dto.dart
class GroupSettingDTO {
  final int? id; // Khóa chính, tự động tăng
  final int conversationId; // Khóa ngoại liên kết với Conversations
  final bool allowMemberInvite;
  final bool allowMemberEdit;
  final int createdBy; // Khóa ngoại liên kết với Users
  final DateTime createdAt; // Thời gian tạo, mặc định là hiện tại
  final bool allowMemberRemove;

  GroupSettingDTO({
    this.id,
    required this.conversationId,
    required this.allowMemberInvite,
    required this.allowMemberEdit,
    required this.createdBy,
    DateTime? createdAt,
    required this.allowMemberRemove,
  }) : createdAt = createdAt ?? DateTime.now();

  factory GroupSettingDTO.fromJson(Map<String, dynamic> json) {
    return GroupSettingDTO(
      id: json['id'],
      conversationId: json['conversationId'],
      allowMemberInvite: json['allowMemberInvite'],
      allowMemberEdit: json['allowMemberEdit'],
      createdBy: json['createdBy'],
      createdAt: DateTime.parse(json['createdAt']),
      allowMemberRemove: json['allowMemberRemove'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'conversationId': conversationId,
      'allowMemberInvite': allowMemberInvite,
      'allowMemberEdit': allowMemberEdit,
      'createdBy': createdBy,
      'createdAt': createdAt.toIso8601String(),
      'allowMemberRemove': allowMemberRemove,
    };
  }

  bool checkGroupSetting() {
    if (conversationId <= 0 || createdBy <= 0) {
      throw ArgumentError('ConversationId and CreatedBy must be positive integers');
    }
    return true;
  }
}

File: server/DTO/GroupSettingDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.Json.Serialization;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace server.DTO
{
    public class GroupSettingDTO
    {
        
        [JsonPropertyName("id")]
        public int? Id { get; set; } // Khóa chính, tự động tăng
        [JsonPropertyName("conversationId")]
        public int ConversationId { get; set; } // Khóa ngoại liên kết với Conversations
        [JsonPropertyName("allowMemberInvite")]
        public bool AllowMemberInvite { get; set; }
        [JsonPropertyName("allowMemberEdit")]
        public bool AllowMemberEdit { get; set; }
        [JsonPropertyName("createdBy")]
        public int CreatedBy { get; set; } // Khóa ngoại liên kết với Users
        [JsonPropertyName("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.Now; // Thời gian tạo, mặc định là hiện tại
        [JsonPropertyName("allowMemberRemove")]
        public bool AllowMemberRemove { get; set; }

        
        public bool CheckGroupSetting(GroupSettingDTO groupSetting)
        {
            if (groupSetting == null)
            {
                throw new ArgumentNullException(nameof(groupSetting), "GroupSettingDTO cannot be null");
            }

            // Kiểm tra các trường bắt buộc
            if (groupSetting.ConversationId <= 0 || groupSetting.CreatedBy <= 0)
            {
                throw new ArgumentException("ConversationId and CreatedBy must be positive integers", nameof(groupSetting));
            }

            return true;
        }

    }
}